src/marqo/tensor_search/tensor_search.py

```diff
@@ -293,6 +295,10 @@ def _add_documents_unstructured(config: Config, add_docs_params: AddDocsParams,
                     # Generate exactly 1 chunk with the custom vector.
                     chunks = [f"{field}::{copied[field]['content']}"]
                     embeddings = [copied[field]["vector"]]
+                    # If normalize_embeddings is true and the index version is > 2.12.0, normalize the embeddings.
+                    # We have added version specific check here to prevent backwards compatibility issues.
+                    if marqo_index.normalize_embeddings and marqo_index.parsed_marqo_version() > semver.VersionInfo.parse(constants.MARQO_CUSTOM_VECTOR_NORMALIZATION_MINIMUM_VERSION):
```

Comment:
Better to parse and store as a class variable than to parse every time

Reply:
makes sense, I'll change this.

Reply:
Changed: https://github.com/marqo-ai/marqo/pull/970#diff-d562ff1ee5c3bc3e20805164a98160b80530cfca5b9a56d8d3020a6c5e575518R17-R19